CMakeLists.txt

```diff
@@ -0,0 +1,90 @@
+# Copyright 2016-2021 Lawrence Livermore National Security, LLC and other
+# IREP Project Developers. See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: MIT
+
+# To use this cmakefile Lua needs to be installed or add Lua's prefix to CMAKE_PREFIX_PATH
+
+cmake_minimum_required(VERSION 3.1)
+project(irep)
+
+# Find lua and figure out bin dir
+find_package(Lua REQUIRED)
+string(REGEX REPLACE "/include$" "/bin" LUA_BIN ${LUA_INCLUDE_DIR})
+
+# Set up IREP_GENERATE varaible
+include(${CMAKE_CURRENT_SOURCE_DIR}/share/irep/irep-config.cmake)
+
+# Be sure to include lua headers
+include_directories(
+    ${CMAKE_CURRENT_SOURCE_DIR}
+    ${LUA_INCLUDE_DIR}
+)
+
+# ir_std.f & ir_extern.f is generated from coresponding
+# headers using IREP_GENERATE
+add_custom_command(
+    OUTPUT ir_std.f
+    COMMAND
+        ${CMAKE_COMMAND} -E env PATH="${LUA_BIN}:$ENV{PATH}"
+        ${IREP_GENERATE} --mode fortran ${CMAKE_CURRENT_SOURCE_DIR}/ir_std.h > ir_std.f
+)
+
+add_custom_command(
+    DEPENDS ir_std.f
+    OUTPUT ir_extern.f
+    COMMAND
+        ${CMAKE_COMMAND} -E env PATH="${LUA_BIN}:$ENV{PATH}"
+        ${IREP_GENERATE} --mode fortran ${CMAKE_CURRENT_SOURCE_DIR}/ir_extern.h > ir_extern.f
+)
+
+# libIR.a is mostly basic functions from irep.c.
+# ir_extern.f, & ir_std.f are very much like wkt headers.
+# This ensures their tables are in libIR.a. Note that they are automatically included
+# in index libraries by irep-config.cmake
+add_library(
+    IR STATIC
+    irep.c
+    ${CMAKE_CURRENT_BINARY_DIR}/ir_extern.f
+    ${CMAKE_CURRENT_BINARY_DIR}/ir_std.f
+)
+
+# Copy libIR.a to current source directory
+add_custom_command(
+    TARGET IR
+    POST_BUILD
+    COMMAND
+        ${CMAKE_COMMAND} -E copy libIR.a ${CMAKE_INSTALL_PREFIX}/libIR.a
+)
+
+# Install irep-generate in bin dir
+install(
+    FILES bin/irep-generate
+    DESTINATION ${CMAKE_INSTALL_PREFIX}/bin
+)
+
+# Install libIR.a in lib dir
+install(
+    FILES libIR.a
+    DESTINATION  ${CMAKE_INSTALL_PREFIX}/lib
+)
+
+# Install headers in include dir
+install(
+    FILES ir_end.h ir_extern.h ir_index.h ir_macros.h ir_start.h ir_std.h ir_undef.h
+    DESTINATION  ${CMAKE_INSTALL_PREFIX}/include
+)
+
+# Install fortran modules in include/fortran dir
+install(
+    FILES build/ir_std.f build/ir_extern.f
```

Comment:
These are the Fortran sources but it's actually the compiled module files that we want to install.  You should see them in the build directory after you've built `libIR.a`.  So I believe what you want is:

```cmake
install(
    FILES ${CMAKE_CURRENT_BINARY_DIR}/ir_std.mod ${CMAKE_CURRENT_BINARY_DIR}/ir_extern.mod
    ...
)
```

Note that `${CMAKE_CURRENT_BINARY_DIR}` is the right way to refer to the build directory (or subdirectory -- it works in nested `CMakeLists.txt` files.  Using paths like `build/ir_std.f` works here because you decided to call your build directory `build` and it happens to be inside the source directory.  Your user may put it anywhere they want, and it may not be named `build`.